### PR TITLE
🧹 optimise GoCardless bank sync to use fewer api calls

### PR DESCRIPTION
**Pull request by matt-fidd**
Part of actualbudget/actual-server#431
Complimentary PR to actualbudget/actual-server#436

The balance fetched from GoCardless is used to calculate the starting balance, which is only relevant when a completely blank account is synced. This change reduces the amount of API calls per sync, once the account has had a starting balance set, to 1 - maximising the new 10 requests/day rate limit coming on the 19th.

### CODE REVIEW

File: packages/loot-core/src/server/accounts/sync.ts [hunk 1]
```diff
@@ -116,19 +111,31 @@ async function downloadGoCardlessTransactions(
...
+  } else {
+    const {
+      transactions: { all },
+    } = res;
+
+    console.log('Response:', res);
+
+    return {
+      transactions: all,
+    };
```
Comment (psybers) [inline packages/loot-core/src/server/accounts/sync.ts:137]: Is it really necessary to destructure, then immediately restructure here?  Couldn't you just return `res`?

File: packages/loot-core/src/server/accounts/sync.ts [hunk 2]
```diff
@@ -116,19 +111,31 @@ async function downloadGoCardlessTransactions(
...
-    transactions: { all },
-    balances,
-    startingBalance,
-  } = res;
+  if (includeBalance) {
+    const {
+      transactions: { all },
+      balances,
+      startingBalance,
+    } = res;
```
Comment (psybers) [inline packages/loot-core/src/server/accounts/sync.ts:119]: Similar question as below, why not just build what we want to return and then return `res`?

**Review by psybers — CHANGES_REQUESTED** [review]

File: packages/loot-core/src/server/accounts/sync.ts (hunk 2 above)
Comment (matt-fidd) [reply packages/loot-core/src/server/accounts/sync.ts:119]: I prefer it in this case to the alternative:

```js
    return {                                                                    
      transactions: res.transactions.all,                                       
      accountBalance: res.balances,                                             
      startingBalance: res.startingBalance,                                     
    };
```

File: packages/loot-core/src/server/accounts/sync.ts (hunk 1 above)
Comment (matt-fidd) [reply packages/loot-core/src/server/accounts/sync.ts:137]: This one's probably unnecessary.

File: packages/loot-core/src/server/accounts/sync.ts (hunk 2 above)
Comment (matt-fidd) [reply packages/loot-core/src/server/accounts/sync.ts:119]: We also want to log the original `res` out for debugging purposes

**Review by psybers — APPROVED** [review]

**Review by MatissJanis — APPROVED** [review]

**Comment by yoyotogblo** [comment]
Any reason why you removed the accountBalance from SimpleFin transactions? I download that in my instance and add it to the account notes. It's a great way to see what my balance is.

**Comment by matt-fidd** [comment]
> Any reason why you removed the accountBalance from SimpleFin transactions? I download that in my instance and add it to the account notes. It's a great way to see what my balance is and to reconcile my transactions.

Yes, it wasn't used anywhere in the code base.
I'll add it back for simplefin as it doesn't hurt to be there!

**Pull request merged**